zerver/webhooks/front/view.py

```diff
@@ -0,0 +1,188 @@
+
+    # Comment or mention
+    elif event_type == 'comment' or event_type == 'mention':
+        name, comment = get_source_name(payload), get_comment(payload)
+        if not (name and comment):
+            return json_error(_("Missing required data"))
+
+        body = "**{name}** left a comment:\n```quote\n{comment}\n```" \
+            .format(name=name, comment=comment)
+
+    # Conversation assigned
+    elif event_type == 'assign':
+        source_name = get_source_name(payload)
+        target_name = get_target_name(payload)
+
+        if not (source_name and target_name):
+            return json_error(_("Missing required data"))
+
+        if source_name == target_name:
+            body = "**{source_name}** assigned themselves." \
+                .format(source_name=source_name)
+        else:
+            body = "**{source_name}** assigned **{target_name}**." \
+                .format(source_name=source_name, target_name=target_name)
+
+    # Conversation unassigned
+    elif event_type == 'unassign':
+        name = get_source_name(payload)
+        if not name:
+            return json_error(_("Missing required data"))
+
+        body = "Unassined by **{name}**.".format(name=name)
+
+    # Conversation archived
+    elif event_type == 'archive':
+        name = get_source_name(payload)
+        if not name:
+            return json_error(_("Missing required data"))
+
+        body = "Archived by **{name}**.".format(name=name)
+
+    # Conversation reopened
+    elif event_type == 'reopen':
+        name = get_source_name(payload)
+        if not name:
+            return json_error(_("Missing required data"))
+
+        body = "Reopened by **{name}**.".format(name=name)
+
+    # Conversation deleted
+    elif event_type == 'trash':
+        name = get_source_name(payload)
+        if not name:
+            return json_error(_("Missing required data"))
+
+        body = "Deleted by **{name}**.".format(name=name)
+
+    # Conversation restored
+    elif event_type == 'restore':
+        name = get_source_name(payload)
+        if not name:
+            return json_error(_("Missing required data"))
+
+        body = "Restored by **{name}**.".format(name=name)
+
+    # Conversation tagged
+    elif event_type == 'tag':
+        name, tag = get_source_name(payload), get_tag(payload)
+        if not (name and tag):
+            return json_error(_("Missing required data"))
+
+        body = "**{name}** added tag **{tag}**.".format(name=name, tag=tag)
+
+    # Conversation untagged
+    elif event_type == 'untag':
+        name, tag = get_source_name(payload), get_tag(payload)
+        if not (name and tag):
+            return json_error(_("Missing required data"))
+
+        body = "**{name}** removed tag **{tag}**.".format(name=name, tag=tag)
+    else:
+        return json_error(_("Unknown webhook request"))
+
+    check_send_stream_message(user_profile, request.client, stream, topic, body)
+
+    return json_success()
```

Comment:
Okay, so there is a better way of structuring your functions here. I have a couple of points (that have larger relevance that I will explain at the end of the comment):

1. Instead of setting `body` here, your helper functions such as `get_tag` should return the formatted body. So maybe name it to `get_tag_body` or something. See how [our GitLab integration does this](https://github.com/zulip/zulip/blob/master/zerver/webhooks/gitlab/view.py#L64).
2. We should probably move all of these `if/elif` clauses to separate function `event` functions, like [our GitLab integration does here](https://github.com/zulip/zulip/blob/master/zerver/webhooks/gitlab/view.py#L64).
3. Also, look at how concise the [main function for our GitLab integration is](https://github.com/zulip/zulip/blob/master/zerver/webhooks/gitlab/view.py#L270). I would suggest restructuring this webhook in a similar fashion, with `EVENT_FUNCTION_MAPPER` and `get_event`, etc. :)

So, there are a few reasons for why all of the above is important:
1. In a webhook with many event types, every event should have a self-contained function that takes care of it. That makes it easier to maintain and to quickly change things. It is a lot more difficult to wade through so many if/else statements to find the one you are looking for.
2. Most of our webhooks are implemented in a similar way, I think we should mimic their style whenever possible for consistency.
3. One near-future project I am hoping to get started on is to implement a UI for our users to modify which specific payloads they would like to be notified of. That is much easier to do with the kind of structure I suggested above because all I would have to do to get all events and functions is to get the `EVENT_FUNCTION_MAPPER` dictionary. :)

Sorry for rambling on, I should probably put all of this in a style guide somewhere. But I hope it helped! :)